main.tf

```
@@ -472,3 +472,105 @@ resource "aws_autoscaling_schedule" "this" {
   # Cron examples: https://crontab.guru/examples.html
   recurrence = lookup(each.value, "recurrence", null)
 }
+
+################################################################################
+# Target scaling policy schedule
+################################################################################
+resource "aws_autoscaling_policy" "this" {
+  for_each = var.create_asg && var.create_scaling_policy ? var.scaling_policies : {}
```

I haven't tested this and it may work, but I suspect we might have to do:

```suggestion
  for_each = { for k,v in var.scaling_policies : k => v if var.create_asg && var.create_scaling_policy }
```